code/datums/storage/storage.dm

```diff
@@ -0,0 +1,1027 @@
+/*!
+ * Datumized Storage
+ * Eliminates the need for custom signals specifically for the storage component, and attaches a storage variable (atom_storage) to every atom.
+ */
+
+///Helper proc to give something storage
+/atom/proc/create_storage(storage_type = /datum/storage, list/canhold, list/canthold, list/storage_type_limits)
+	RETURN_TYPE(/datum/storage)
+
+	if(atom_storage)
+		QDEL_NULL(atom_storage)
+
+	atom_storage = new storage_type(src)
+
+	if(canhold || canthold)
+		atom_storage.set_holdable(canhold, canthold, storage_type_limits)
+
+	return atom_storage
+
+// The parent and real_location variables are both weakrefs, so they must be resolved before they can be used.
+/datum/storage
+	/**
+	 * A reference to the atom linked to this storage object
+	 * If the parent goes, we go. Will never be null.
+	 */
+	VAR_FINAL/atom/parent
```

Comment:
Why all the var final ?

Reply:
Parent is set by the proc set_parent()
We don't want people to override this var on subtypes, it should only be handled by the code in the file.
Same with our proc set_holdable()